### PR TITLE
Fix forceprojectkeys

### PR DESCRIPTION
**Pull request by M0ses**

### CODE REVIEW

**Comment by M0ses** [comment]
added "DO NOT MERGE" because this draft is not tested yet - I wil build an appliance in my home and test it.

File: dist/setup-appliance.sh [hunk 1]
```diff
@@ -643,12 +643,20 @@ function prepare_os_settings {
...
+    return
+  fi
+
+  # This is done manually and not via api to avoid authentication
+  # and service dependency (systemd) problems.
   cd $apidir
-  mysql -e "update configurations SET enforce_project_keys=$ENABLE_FORCEPROJECTKEYS" api_production
-  RAILS_ENV=production bin/rails writeconfiguration
+  echo " - Setting enforce_project_keys in api_production.configurations to '$ENABLE_FORCEPROJECTKEYS'"
+  mysql -u $MYSQL_USER -p $MYSQL_PASS -e "update configurations SET enforce_project_keys=$ENABLE_FORCEPROJECTKEYS" api_production || exit 1
```
Comment (eduardoj) [inline dist/setup-appliance.sh:656]: Better to remove the space in between `-p` and `$MYSQL_PASS`?

File: dist/setup-appliance.sh [hunk 2]
```diff
@@ -638,11 +639,44 @@ function prepare_os_settings {
...
+
+  # This is done manually and not via api to avoid authentication
+  # and service dependency (systemd) problems.
+  cd $apidir
+  echo " - Setting enforce_project_keys in api_production.configurations to '$ENABLE_FORCEPROJECTKEYS'"
+  mysql -u $MYSQL_USER -p$MYSQL_PASS -e "update configurations SET enforce_project_keys=$ENABLE_FORCEPROJECTKEYS" api_production || exit 1
+
+  echo " - Starting 'rails writeconfiguration'"
+  RAILS_ENV=production bin/rails writeconfiguration || exit 1
+}
```
Comment (mlschroe) [inline dist/setup-appliance.sh:666]: This needs running source/repo servers...

File: dist/setup-appliance.sh [hunk 3]
```diff
@@ -654,6 +688,9 @@ export LC_ALL=C
 prepare_os_settings
 
 ENABLE_OPTIONAL_SERVICES=0
+ENABLE_FORCEPROJECTKEYS=1
```
Comment (mlschroe) [inline dist/setup-appliance.sh:691]: I think this needs discussion. I'm kind of opposed to have forceprojectkeys enabled as a default.

File: dist/setup-appliance.sh (hunk 2 above)
Comment (mlschroe) [reply dist/setup-appliance.sh:666]: OTOH I don't even know how the initial writethrough of the configuration works at all...

Comment (M0ses) [reply dist/setup-appliance.sh:666]: Should be started at this point - setting up database initially also does a `writeconfiguration`

File: dist/setup-appliance.sh (hunk 3 above)
Comment (M0ses) [reply dist/setup-appliance.sh:691]: No strong opinions here - I leave it up to you and @adrianschroeter to decide the default behaviour.
This has been a "regression" (which is in the code now for 7 years), so I tried to implement the behaviour like before.

And if you would ask me if is smart to "fix" and 7 year old regression - my clear answer - NO

So if you and @adrianschroeter would agree, I would change that.

**Comment by M0ses** [comment]
@hennevogel : I hope that https://github.com/openSUSE/open-build-service/pull/9864/commits/62aa6ee0dc22b94d12488b36ec84b3fc36a4aadf fixes #9779.

I hope that it does not break future database migration (the way I implemented it).

**Comment by M0ses** [comment]
Last but not least - This PR was tested successfully in kanku

**Comment by M0ses** [comment]
Merging - as discussed with @mlschroe and @adrianschroeter

**Pull request merged**